site/docs/hosting/comparison.md

```diff
@@ -0,0 +1,62 @@
+| Deno Deploy           | Free       | No paid plans yet                     | [100k req/day, 1000 req/min, 50ms time limit](https://deno.com/deploy/docs/pricing-and-limits)     | :x:                | :white_check_mark: | :x:                | Beta                                 |
+| Cloudflare Workers    | Free       | $5/10M req                            | [100k req/day, 10ms limit](https://workers.cloudflare.com/)                                        | :x:                | :x:                | :white_check_mark: |                                      |
+| Vercel Edge Functions | Free       | $20/mo subscription for 500K          | [100k req/day](https://vercel.com/pricing)                                                         | :question:         | :question:         | :question:         |                                      |
+| Heroku                | Free       | It's complicated                      | [550-1000 hours per month](https://www.heroku.com/pricing)                                         | :white_check_mark: | :question:         | :question:         | Long startup times, not recommended? |
+| serverless.com        | Free       |                                       |                                                                                                    | :question:         | :question:         | :question:         |                                      |
+| DigitalOcean Apps     | $5         |                                       |                                                                                                    | :question:         | :question:         | :question:         | Not tested                           |
+| Fastly Compute@Edge   |            |                                       |                                                                                                    |                    |                    |                    |                                      |
+
+### VPS
+
+A virtual private server is a virtual machine that you have full control over.
+You can usually access it via [SSH](https://en.wikipedia.org/wiki/Secure_Shell).
+You can install any software there, and are responsible for system updates and so on.
+
+You can run on bot on both polling or webhooks on a VPS.
+
+Check out [the tutorial](./vps.md) on how to host a grammY bot on a VPS.
+
+| Name          | Min. price | Ping to API                               | Cheapest option                    |
+| ------------- | ---------- | ----------------------------------------- | ---------------------------------- |
+| DigitalOcean  | $5         | 1-15 ms :netherlands: AMS, 19 ms :de: FRA | 1 vCPU, 1 GB RAM, 25 GB SSD, 1 TB  |
```

Comment:
```suggestion
| DigitalOcean  | $5         | 1-15ms :netherlands: AMS, 19 ms :de: FRA  | 1 vCPU, 1 GB RAM, 25 GB SSD, 1 TB  |
```

Reply:
https://grammy.dev/plugins/session.html#external-storage-solutions

Reply:
Units of measurement should always have a space character. The only exception where this isn't quite clear is the per cent sign.

Reply:
How do they relate?

Reply:
Should we replace the other `ms` and `s` to have a space before them?

Reply:
Which ones? I don't see any of them.

Reply:
Wrong column. Found them.

Reply:
Fixed.